### DISCUSSION TITLE
E-mail templates docs

### DISCUSSION THREAD

**Discussion by iBobik** [contember/admin#632] [Support]
Hi, I want to customize e-mail templates and haven't found docs about it.

<img width="897" alt="Snímek obrazovky 2023-09-26 v 18 32 19" src="https://github.com/contember/interface/assets/614232/31d3389a-345f-4b47-91d4-25df7b354fa7">

- How to see existing templates? It would be helpful for creating customized one.
- What placeholders can I use and what templating language is there?
- How to use variants?
- What is layout and how to use it?

And my feature requests :-)
- Ability to use it for any other emails in my app - custom type and send mutation.
- Define templates in a code, so easy deployment.

**Comment by matej21** [comment]
Hi, there's an unmerged [PR](https://github.com/contember/docs/pull/140) for the docs. You can review the documentation preview here: https://mail-templates.docs-dwe.pages.dev/reference/engine/tenant/mail-templates. I hope that this will answer all your questions. :)

Regarding the two feature requests:
- Custom emails: We might consider this, although I'm not sure if it's within our scope.
- Templates as code: That's definitely a good idea. While you could potentially make API calls during deployment to set up the templates, I agree that some additional tooling would be beneficial.
